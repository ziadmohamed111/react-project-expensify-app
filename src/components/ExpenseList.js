import React from 'react'
import { connect } from "react-redux"
import ExpenseListItem from './ExpenseListItem'
import selectExpense from "../selectors/expenses"

export const ExpenseList = ( props ) => (
    <div className="content-container">
        <div className="list-header">
            <div className="show-for-mobile">Expenses</div>
            <div className="show-for-desktop">Expense</div>
            <div className="show-for-desktop">Amount</div>
        </div>
        {
            props.expenses.length === 0 ? (
                <div className="list-item list-item--message">
                    <span>No expenses</span>
                </div>
            ) : ( 
                props.expenses.map( ( expense ) => <ExpenseListItem key={expense.id} {...expense} /> ) 
                )

    }
    </div>
);
const mapStateIntoProps = ( state ) => {
    return {
        expenses: selectExpense( state.expenses , state.filters )
    }
}

export default connect( mapStateIntoProps )( ExpenseList )
