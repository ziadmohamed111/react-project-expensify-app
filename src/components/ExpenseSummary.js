import React from 'react';
import {connect} from "react-redux"
import totalExpenses from "./totalExpenses"
import numeral from "numeral"

export const ExpenseSummary = (props) => (
    <div>
        <p>Viewing {props.expenses.length} {props.expenses.length === 1 ? "expense" : "expenses" } totalling {numeral( totalExpenses( props.expenses ) / 100 ).format( '$0,0.00' )} </p>
    </div>
)

const mapStateToProps = ( state ) => ( {
    expenses: state.expenses
} );

export default connect( mapStateToProps )( ExpenseSummary)