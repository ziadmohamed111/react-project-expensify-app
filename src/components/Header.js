import React from "react";
import { NavLink } from "react-router-dom"

const Header = () => (
    <header>
        <h1>Expensify</h1>
        <NavLink to="/" activeClassName="is-active" exact={true}> DashBoard </NavLink>
        <NavLink to="/create" activeClassName="is-active"> Go create </NavLink>
    </header>
)

export default Header