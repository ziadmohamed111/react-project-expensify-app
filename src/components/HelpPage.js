import React from 'react';

export const HelpPage = () => (
    <div>
        <p>HelpPage</p>
    </div>

)

export default HelpPage