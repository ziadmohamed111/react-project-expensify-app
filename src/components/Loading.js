import React from "react";

const Loading = ()=>(
    <div className="Loading-Page">
        <div className="loading">
            <div className="loading-div"></div>
            <div className="loading-div"></div>
            <div className="loading-div"></div>
            <div className="loading-div"></div>
            <div className="loading-div"></div>
            <div className="loading-div"></div>
            <div className="loading-div"></div>
            <div className="loading-div"></div>
        </div>
    </div>
)
export default Loading