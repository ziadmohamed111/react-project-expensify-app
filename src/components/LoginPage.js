import React from "react"
import { connect } from "react-redux"
import { startLogin } from "../actions/auth"

export const LoginPage = ( props ) => (
    <div className="box-layout">
        <div className="box-layout__login-box">
            <h1>Expensify</h1>
            <p>It's time to get your expenses under control.</p>
            <button className="button-big" onClick={props.startLogin}>Login With Google</button>
        </div>
        
    </div>
)

const mapDispatchToProps = ( dispatch ) => ( {
    startLogin: () => dispatch( startLogin() )
} )

export default connect( undefined, mapDispatchToProps )( LoginPage ) 