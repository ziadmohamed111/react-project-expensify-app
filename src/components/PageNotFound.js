import React from 'react';
import { Link } from "react-router-dom" 

export const PageNotFound = () => (
    <div>
        <p>404!</p>
        <Link to="/">GO HOME</Link>
    </div>

)

export default PageNotFound