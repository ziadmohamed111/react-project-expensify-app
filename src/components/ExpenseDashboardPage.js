import React from 'react';
import ExpenseList from "./ExpenseList"
import ExpenseListFilters from "./ExpenseListFilters"

export const ExpenseDashboardPage = () => {    
    return (
        (
            <div>
                <ExpenseListFilters/>
                <ExpenseList/>
            </div>
        )
    )
}

export default ExpenseDashboardPage