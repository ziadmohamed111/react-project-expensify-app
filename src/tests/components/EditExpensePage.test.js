import React from "react"
import { shallow } from "enzyme"
import { EditExpensePage } from "../../components/EditExpensePage"
import expenses from "../fixtures/expenses"

let history, editExpense, removeExpense, wrapper

beforeEach( () => {
    editExpense = jest.fn()
    removeExpense = jest.fn()
    history = { push: jest.fn() }
    wrapper = shallow(
        <EditExpensePage expense={expenses[1]} removeExpense={removeExpense} editExpense={editExpense} history={history} /> )
} )

test( "should render edit Expense Page Correctly", () => {
    expect( wrapper ).toMatchSnapshot()
} )

test( "should handel on Submit", () => {
    wrapper.find( "ExpenseForm" ).prop( "onSubmit" )( expenses[1] )
    expect( history.push ).toHaveBeenLastCalledWith( "/" )
    expect( editExpense ).toHaveBeenCalledWith( expenses[1].id, expenses[1] )
} )

test( "should handel on Remove", () => {
    wrapper.find( "button" ).simulate( "click" )
    expect( history.push ).toHaveBeenLastCalledWith( "/" )
    expect( removeExpense ).toHaveBeenCalledWith( { id: expenses[1].id } )
} )


