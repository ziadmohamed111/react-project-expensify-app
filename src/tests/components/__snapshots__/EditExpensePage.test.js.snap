// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render edit Expense Page Correctly 1`] = `
<div>
  <ExpenseForm
    expense={
      Object {
        "amount": 100000,
        "createdAt": -345600000,
        "description": "Rent",
        "id": "2",
        "note": "",
      }
    }
    onSubmit={[Function]}
  />
  <button
    onClick={[Function]}
  >
    Remove
  </button>
</div>
`;
